daffodil-lib/src/main/scala/org/apache/daffodil/lib/xml/XMLUtils.scala

```diff
@@ -850,6 +886,14 @@ Differences were (path, expected, actual):
     arrayCounters
   }
 
+  private def getXSIType(a: Elem) = {
+    val res = a.attribute(XSI_NAMESPACE.toString, "type").map(_.head.text).orElse {
+      val a2 = a.attributes.asAttrMap.get("xsi:type")
```

Comment:
This `a2` is new. Is this because `xmlns:xsi` isn't define in some infosets? Should we just define them so they are valid XML?

Reply:
Removed. Requires an xmlns:xsi definition now.